alternator/streams.cc

```diff
@@ -140,7 +140,7 @@ namespace alternator {
 future<alternator::executor::request_return_type> alternator::executor::list_streams(client_state& client_state, service_permit permit, rjson::value request) {
     _stats.api_operations.list_streams++;
 
-    auto limit = rjson::get_opt<int>(request, "Limit").value_or(std::numeric_limits<int>::max());
+    auto limit = rjson::get_opt<int>(request, "Limit").value_or(100);
```

Comment:
An interesting question is what happens on DynamoDB if you try to set a limit higher than 100. As usual, I'll write a test to check and let you know :-)

Reply:
As I understood from the documentation, 100 is not only DynamoDB's default limit, it is also the maximum allowed. If you try 101 it complains with a ValidationException: "One or more parameter values were invalid: Limit cannot be greater than the maximum allowed size: 100".

A test for this:
```python
def test_list_streams_too_high_limit(dynamodb, dynamodbstreams):
    with create_stream_test_table(dynamodb, StreamViewType='KEYS_ONLY') as table:
        with pytest.raises(ClientError, match='ValidationException'):
            wait_for_active_stream(dynamodbstreams, table)
            dynamodbstreams.list_streams(Limit=101)

```

If you want to make this PR even better, perhaps you can do that?

Reply:
I considered it, but then we should change some other places which allow much higher limits than Dynamo as well. And this in turn would break our dtests etc which apparently use limits of 1000 or so. Probably to get around the insane number of shards Scylla generates and all the round tripping it entails. I don't think we really need to ensure _limitations_ are equal to AWS do we? Just defaults...

Reply:
Anyway, never mind about this now. I'll send a separate patch for testing (and/or changing) this aspect.

Reply:
Sent https://github.com/scylladb/scylladb/pull/12625